frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,367 @@
+                e.map(dispatcher =>
+                    setDispatchers(preState => ([
+                        ...preState, dispatcher
+                    ])));
+            }
+        }
+    };
+
+    const handleIdentifier = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            identifier: e.target.value
+        }));
+    };
+
+    const handleType = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            type: e
+        }));
+        setDropdownMenuName(e);
+    };
+
+    const handleCity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, city: e.target.value}
+        }));
+    };
+
+    const handleLineOne = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine1: e.target.value}
+        }));
+    };
+
+    const handleLineTwo = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine2: e.target.value}
+        }));
+    };
+
+    const handleTotalCapacity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            totalCapacity: e.target.value
+        }));
+    };
+
+    const addWarehouseHandler = (e) => {
+        e.preventDefault();
+
+        let updateWarehouseDto = {};
+        if (dispatchers.length) {
+            const dispatchersId = dispatchers.map(dispatcher => dispatcher.id);
+            updateWarehouseDto = {
+                ...warehouseDto,
+                dispatchersId: dispatchersId,
+                customerId: props.currentCustomerId
+            }
+        } else {
+            updateWarehouseDto = {
+                ...warehouseDto,
+                dispatchersId: [],
+                customerId: props.currentCustomerId
+            }
+        }
+
+        const validationResult = validateWarehouse(updateWarehouseDto, dropdownMenuName,
+            updateWarehouseDto.dispatchersId);
+        setErrors(preState => ({
+            ...preState,
+            validationErrors: validationResult,
+            serverErrors: ''
+        }));
+
+        if (!validationResult.includes("identifier")) {
+            fetch(`/customers/${props.currentCustomerId}/warehouses/identifier?identifier=${updateWarehouseDto.identifier}`)
+                .then(resp => resp.json())
+                .then(isIdentifierExist => {
+
+                    const validationResult = validateWarehouseWithIdentifierExist(updateWarehouseDto, dropdownMenuName,
+                        updateWarehouseDto.dispatchersId, isIdentifierExist);
+                    setErrors(preState => ({
+                        ...preState,
+                        validationErrors: validationResult,
+                        serverErrors: ''
+                    }))
+                })
+        }
+        if (!validationResult.length) {
```

Comment:
Will validationResult be waiting for the response from previous block?

Reply:
yes, thanks!